crates/symbolicator/src/services/download/http.rs

```diff
@@ -61,37 +61,74 @@ impl HttpDownloader {
         &self,
         file_source: HttpRemoteDif,
         destination: PathBuf,
+    ) -> Result<DownloadStatus, DownloadError> {
```

Comment:
This needs a bit of an explanation.

In all non-Sentry, non-local FS sources only the initial GET request is/was being retried. In contrast, downloads from Sentry sources retry both the initial GET request and the stream attempts, which is what the diff in this file attempts to mimic. See: 
https://github.com/getsentry/symbolicator/blob/919aafb2abda6f150d8904b6958ad7b3fce71757/crates/symbolicator/src/services/download/sentry.rs#L236-L314

Is there any reason why the others (HTTP, S3, GCS) aren't retrying both the initial GET request as well as the stream?

Reply:
I've removed retry-related logic in this PR to keep it focused. A new PR branched off of the changes in this one has been opened which extends retrying on both the header and the stream to all sources: https://github.com/getsentry/symbolicator/pull/485. It's a pretty naive approach right now, but I'd love your feedback on the diff there as well @flub.